tests/vectordbs/test_qdrant.py

```diff
@@ -0,0 +1,112 @@
+"""Tests for verifying if the `qdrant` vector DB server is up and running."""
+
+import docker
+import numpy as np
+import pytest
+from qdrant_client import QdrantClient
+from qdrant_client.models import Distance, PointStruct, VectorParams
+
+import inatinqperf
+
+
+@pytest.fixture(scope="module", autouse=True)
+def container_fixture():
+    """Start the docker container with the vector DB."""
+    client = docker.from_env()
+    container = client.containers.run(
+        "qdrant/qdrant",
+        ports={"6333": "6333"},
+        remove=True,
+        detach=True,  # enabled so we get back a Container object
+    )
+
+    yield container
+    container.stop()
+
+
+@pytest.fixture(name="client")
+def client_fixture():
+    """Fixture to get QdrantClient."""
+    # Connect to existing Qdrant instance
+    client = QdrantClient("http://localhost:6333")
+    return client
+
+
+@pytest.fixture(name="client_with_collection")
+def client_with_collection_fixture(client: QdrantClient):
+    """Fixture to yield a client with a collection initialized."""
+    collection_name = "test_collection"
+    client.create_collection(
+        collection_name=collection_name,
+        vectors_config=VectorParams(size=100, distance=Distance.COSINE),
+    )
+    yield client
+    client.delete_collection(collection_name)
+
+
+@pytest.fixture(name="rng")
+def rng_fixture():
+    """Fixture for a random number generator with a fixed seed."""
+    rng = np.random.default_rng(seed=101)
+    return rng
+
+
+def test_create_collection(client: QdrantClient, collection_name: str):
+    """Test collection creation."""
+    client.create_collection(
+        collection_name=collection_name,
+        vectors_config=VectorParams(size=100, distance=Distance.COSINE),
+    )
+    assert client.collection_exists(collection_name)
+
+    # Clean up collection
+    client.delete_collection(collection_name)
+
+
+def test_vector_insertion(
+    client_with_collection: QdrantClient,
+    rng: np.random.Generator,
+    collection_name: str,
+):
+    """Test insertion of vectors into the database."""
+    num_vectors = 117
```

Comment:
Is the number `117` significant in any way? Maybe add a comment to suggest that it is just a number or information to explain its significance.

Reply:
Added a comment stating it is not of any significance.

PS I use 117 as a little easter egg in reference to Master Chief from Halo. His full title is `Master Chief Petty Officer John-117`.